src/Enzo/enzo_EnzoPhysicsCosmology.hpp

```diff
@@ -180,48 +180,110 @@ class EnzoPhysicsCosmology : public Physics {
   double length_units() const
   {
     return enzo_constants::Mpc_cm*comoving_box_size_/hubble_constant_now_/
       (1.0 + current_redshift_);
   }
 
-  /// Return current time units (requires set_current_time())
+  /*
+    Return time unit in terms of seconds (requires set_current_time())
+
+    Time unit is defined so that the following is true (which has the effect of simplifying
+    Poisson's equation):
+
+    `4 * pi * G * rho_bar_m_com * time_unit^2 / a_unit^3 = 1`,
+
+    Where `G` is the gravitational constant.`rho_bar_m_com` is the mean comoving matter density
+    of the universe, `time_unit` is the time unit in seconds, and `a_unit` is the "unit 
+    cosmological scale factor". To understand where the factor of `a_unit^3` come from, see
+    Equations 8, 17, 18 of Greg L. Bryan et al 2014 ApJS 211 19, and note that Laplacian(`phi`) 
+    has dimensions of `time^(-2) * a^(2).
+
+    In Enzo-E, `a_unit` is defined so that `a` is 1 at the initial redshift (`z_i`), so that:
+    `(1+z_i)^(-1) / a_unit = 1`, which means that `a_unit = 1 / (1 + z_i)`.
+
+    `rho_bar_m_0` can be written as `Omega_m_0 * rho_crit_0`, where `Omega_m_0` is a cosmological
+    parameter which can be set in the input parameter file,
+    and `rho_crit_0` is defined as `3 * H_0^2 / (8 * pi * G)`, where `H_0` is the expansion rate
+    of the universe at redshift 0 (with dimensions of inverse time). Plugging this all in gives:
+
+    `3 / 2 * Omega_m_0 * H_0^2 * time_unit^2 * (1 + z_i)^3 = 1`.
+
+    After some rearrangement, we get the following expression:
+
+    `time_unit = sqrt(2/3) * h / H0 / (h * sqrt(Omega_m_0 * (1 + z)^3))`/
+
+  */
+
+
   double time_units() const
   {
-    return 2.519445e17/sqrt(omega_matter_now_)/hubble_constant_now_/
-      pow(1.0 + initial_redshift_,1.5);
+    return sqrt(2/3) / (enzo_constants::H0_over_h * hubble_constant_now_ *
+			sqrt(omega_matter_now_ *
+			     (1 + initial_redshift_) *
+			     (1 + initial_redshift_) *
+			     (1 + initial_redshift_)));
   }
 
-  double velocity_units() const
-  {
-    return 1.22475e7*comoving_box_size_*sqrt(omega_matter_now_)*
-                      sqrt(1.0 + initial_redshift_);
-  }
+  // Return velocity unit in cm/s
+  // Equation 17 of Greg L. Bryan et al 2014 ApJS 211 19, shows that the dimensions of velocity
+  // are `a` times `comoving length` over `time`.
 
-  /// Returns the scaling factor to be divided by temperature to convert from
-  /// units of Kelvin to units of specific internal energy
-  ///
-  /// The original Enzo refered to this quantity as "temperature units", but in
-  /// Enzo-E we primarily track temperature in units of Kelvin
-  double kelvin_per_energy_units() const
+  double velocity_units() const
   {
-    return 1.81723e6*pow(comoving_box_size_,2.0)*omega_matter_now_*
-                      (1.0 + initial_redshift_);
+    return length_units() / time_units() * (1 + initial_redshift_) / (1 + current_redshift_);
```

Comment:
I think this definition is fine and is certainly clearer. The one potential downside is that it is not immediately obvious from inspection that velocity units are independent of time. I think it would be worth adding this as a comment. The other potential issue is that the exact value returned by this routine may change by very small amounts of time due to roundoff error (the 1+z terms will not cancel exactly) -- I don't think this is a problem but just wanted to point it out.